### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by chaitu2512**

### CODE REVIEW

File: entry/src/main/ets/MainAbility/app.ets [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*
+>>>>>>> b0fbad399365034a67a794683fda25e0f20490a6
```
Comment (HMOS-Developer) [inline entry/src/main/ets/MainAbility/app.ets]: remove this line

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+@Component
+struct Index {
+  private model: CardModel = new CardModel('Title', 'SubTitle')
+  private subHead: CardModel = new CardModel('Title', 'Subtitle goes here')
+  private squareOutlinedCard: CardModel = new CardModel('Breaking Podcast', 'Size of Song')
+  private rightImage: CardModel = new CardModel('Play Relaxing Songs', 'Song Name')
+  private rightImageFilledCard: CardModel = new CardModel('Movie Name', 'SubHead')
+  scroller: Scroller = new Scroller()
+
+  private OnClick(button: button.Model) {
```
Comment (HMOS-Developer) [inline entry/src/main/ets/MainAbility/pages/index.ets]: rename to onButtonClick

File: entry/src/main/ets/MainAbility/pages/index.ets [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    console.log(button.getButtonText())
+  }
+  private cardclick() {
```
Comment (HMOS-Developer) [inline entry/src/main/ets/MainAbility/pages/index.ets]: rename to onCardClick

Comment (HMOS-Developer) [inline entry/src/main/ets/MainAbility/pages/index.ets]: rename to onCardClick

File: material-card/src/button.ets [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      this.isText= isText
+      this.buttonText = buttonText
+      this.buttonTextColor = buttonTextColor
+      this.buttonColor= buttonColor
+      this.fontSize = fontSize
+      this.buttonType =buttonType
+      this.buttonHeight = buttonHeight
+      this.buttonWidth =buttonWidth
+    }
+//    onClick(hi : string) {
```
Comment (HMOS-Developer) [inline material-card/src/button.ets]: remove commented code

File: material-card/src/materialCard.ets [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CardModel, CardType, MediaIconPosition } from './cardModel'
+
+@Component
+export struct MaterialCard {
+  @State private model: CardModel = undefined
+  private cardClick : () => void = undefined
```
Comment (HMOS-Developer) [inline material-card/src/materialCard.ets]: remove this variable

Comment (HMOS-Developer) [inline material-card/src/materialCard.ets]: remove this variable

File: README.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+1. Filled Card (Default)
+2. Outlined Card
+3. Elevated Card
+
+## Download & Install
+
+Install using npm
+```c
+npm i ohos-material-card
```
Comment (HMOS-Developer) [inline README.md]: change this to npm i @ohos/material-card

File: package.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    "directoryLevel": "project"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {
+    "hypium": "^1.0.0",
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/material-card": "file:./material-card"
```
Comment (HMOS-Developer) [inline package.json]: remove dependency from here

File: entry/package.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  "version": "1.0.0",
+  "ohos": {
+    "org": "huawei",
+    "buildTool": "hvigor",
+    "directoryLevel": "module"
+  },
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {}
```
Comment (HMOS-Developer) [inline entry/package.json]: add library dependency here - ""@ohos/material-card": "file:../material-card""

File: entry/build-profile.json5 [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  },
+  "targets": [
+    {
+      "name": "default",
+    },
+    {
+      "name": "ohosTest",
+    }
+  ]
+}
```
Comment (archanaatr) [inline entry/build-profile.json5:13]: delete package-lock.json file

File: entry/hvigorfile.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+// Script for compiling build behavior. It is built in the build plug-in and cannot be modified currently.
+module.exports = require('@ohos/hvigor-ohos-plugin').legacyHapTasks
```
Comment (archanaatr) [inline entry/hvigorfile.js]: Looks copyright need to be added in all js / ets files, please check with Leela and add

File: material-card/package.json [hunk 11]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@ohos/material-card",
+  "version": "1.0.0",
+  "description": "",
+  "main": "src/materialCard.ets",
+  "scripts": {
```
Comment (archanaatr) [inline material-card/package.json]: convert material-card to a module, if hyphen is not support, then change name to materialcard

File: material-card/src/cardModel.ets [hunk 12]
```diff
@@ -0,0 +1,434 @@
...
+  }
+
+  getShadowRadius(): number{
+    return this.shadowRadius
+  }
+}
+
+export class CardModel extends Properties {
+  private title: string | Resource = ""
+  private titleFontSize: number = 25
```
Comment (archanaatr) [inline material-card/src/cardModel.ets]: Mark as dmentions as vp and font sizes as fp in both library and samples

File: entry/build-profile.json5 (hunk 9 above)
Comment (chaitu2512) [reply entry/build-profile.json5:13]: Do you need to delete the file completely?

**Pull request merged**